tests/unit/training/test_training_sae.py

```diff
@@ -37,3 +40,110 @@ def test_TrainingSAE_training_forward_pass_can_scale_sparsity_penalty_by_decoder
             pytest.approx(train_step_output.losses["l1_loss"].detach().item())  # type: ignore
             == 2.0 * feature_acts.norm(p=1, dim=1).mean().detach().item()
         )
+
+
+@pytest.mark.parametrize("architecture", ["standard", "gated", "jumprelu"])
+def test_TrainingSAE_encode_returns_same_value_as_encode_with_hidden_pre(
+    architecture: str,
+):
+    cfg = build_sae_cfg(architecture=architecture)
+    sae = TrainingSAE(TrainingSAEConfig.from_sae_runner_config(cfg))
+    x = torch.randn(32, cfg.d_in)
+    encode_out = sae.encode(x)
+    encode_with_hidden_pre_out = sae.encode_with_hidden_pre_fn(x)[0]
+    assert torch.allclose(encode_out, encode_with_hidden_pre_out)
+
+
+def test_TrainingSAE_initializes_only_with_log_threshold_if_jumprelu():
+    cfg = build_sae_cfg(architecture="jumprelu", jumprelu_init_threshold=0.01)
+    sae = TrainingSAE(TrainingSAEConfig.from_sae_runner_config(cfg))
+    param_names = dict(sae.named_parameters()).keys()
+    assert "log_threshold" in param_names
+    assert "threshold" not in param_names
+    assert torch.allclose(
+        sae.threshold,
+        torch.ones_like(sae.log_threshold.data) * cfg.jumprelu_init_threshold,
+    )
+
+
+def test_TrainingSAE_jumprelu_sae_encoding():
+    cfg = build_sae_cfg(architecture="jumprelu")
+    sae = TrainingSAE.from_dict(cfg.get_training_sae_cfg_dict())
```

Comment:
We have this block in three of the tests, so it might help to extract it to a fixture.

Reply:
I'm not a huge fan of pulling out stuff into fixtures unless it saves a lot of lines of code. Making this into a fixture means you can't just look at the test and understand what it's doing, you also need to scroll to look at the fixture. Since this is just 2 lines of code, I'm rather leave it as is IMO